### PR TITLE
cmdutil: Add OTEL contrib runtime instrumentation

### PR DESCRIPTION
**Pull request by ypaq**
## Rationale

We want the option to collect Golang runtime metrics out of the box with `heroku/x/cmdutil`. Here we introduce a new environment variable `OTEL_ENABLE_RUNTIME_METRICS` that when set to `true` will start the collection of runtime metrics and their reporting through the OTEL pipeline.

## Changes
* Add dependency `go.opentelemetry.io/contrib/instrumentation/runtime@v0.45.0`
* Add config `OTEL_ENABLE_RUNTIME_METRICS`, defaulting to `false`
* Add option function to enable runtime metrics collection
* Update semconv version to `go.opentelemetry.io/otel/semconv/v1.25.0`

## Meta
[W-15906716](https://gus.lightning.force.com/a07EE00001sy18ZYAQ)

### CODE REVIEW

**Review by voidlock — APPROVED** [review]
LGTM

**Review by slizco — APPROVED** [review]
👍

File: go-kit/metrics/provider/otel/provider.go [hunk 1]
```diff
@@ -128,7 +130,14 @@ func New(ctx context.Context, serviceName string, opts ...Option) (xmetrics.Prov
 
 // Start starts the provider's controller and exporter.
 func (p *Provider) Start() error {
-	return nil
+	var err error
+	if p.cfg.enableRuntimeMetrics {
+		err = runtime.Start(
+			runtime.WithMeterProvider(p.meterProvider),
+			runtime.WithMinimumReadMemStatsInterval(p.cfg.collectPeriod),
```
Comment (cgfuh) [inline go-kit/metrics/provider/otel/provider.go:137]: Isn't this interval configurable? defaults seem to be 1 minute (`collectPeriod` has the `DefaultReaderInterval`)

**Review by cgfuh — APPROVED** [review]
I left a question, but it looks good.

File: go-kit/metrics/provider/otel/provider.go (hunk 1 above)
Comment (ypaq) [reply go-kit/metrics/provider/otel/provider.go:137]: Correct, it uses the default, unless you set the `WithCollectPeriod` option.

https://github.com/heroku/x/blob/master/go-kit/metrics/provider/otel/options.go#L47-L53

**Review by aaronromeo — APPROVED** [review]

**Pull request merged**